src/utils/helper.c

```diff
@@ -182,6 +182,17 @@ parse_connstring(char *conninfo, char **hostname, int *port)
     return res;
 }
 
+bool set_state(volatile atomic_bool *state, bool value)
+{
+    bool expected;
+
+    do {
+        expected = !value;
+    } while (!atomic_compare_exchange_weak(state, &expected, value));
```

Comment:
or just use _strong and get rid of the busy loop.

Reply:
Good idea. I did that due to performance reasons but I misunderstood what I read. Rereading it now it's very clear that the strong version is preferable here[1]:

> When a compare-and-exchange is in a loop, the weak version will yield better performance on
> some platforms. When a weak compare-and-exchange would require a loop and a strong one
> would not, the strong one is preferable. 

[1]: Obviously, morning me is smarter than afternoon me. :)